Release/2.0.0 alpha.5

rousso opened the pull request.

bertrand-lorentz committed:
Publish via the new Maven Central Portal

Also use the options of the setup-java action to define the credentials
for the Maven Central Portal, instead of having them in a settings.xml
file.
Move definition of the repository to use for snapshot dependencies to
pom.xml. Having it in settings.xml was not useful, and required action
by every developer.

Update README accordingly.

(cherry picked from commit 202fc1a4619f156d37265923e48cc7eb87cd23ff)

rousso committed:
Parent context predicates are now ignored when computing the context of a EFX template line. This yields correct results when both parent and child contexts have predicates.

bertrand-lorentz committed:
Add failing unit test to demonstrate bug with field in predicate

rousso committed:
Refactor field ID retrieval methods in EfxExpressionTranslatorV2 and template context computation EfxTemplateTranslatorV2.

rousso committed:
Update version to 2.0.0-alpha.5 and set eForms core version to 1.5.0

rousso committed:
Applied to EFX-1 the same fix that was implemented in EFX-2 for the correct contextualisation of XPaths with predicates in view templates.

bertrand-lorentz reviewed (commented):
We would not be able to publish this release in Maven Central, as it does not include the changes in develop that are necessary for this ( see PR #125 )

bertrand-lorentz commented:
I've cherry-picked the commit to update the publication to Maven.

bertrand-lorentz committed:
pom: Set a deployment name when publishing to Maven Central

This will make it easier to identify each release on the Central Portal
"Deployments" page.
